File: BullsAndCows/BullsAndCows.py
class Solution(object):
    def getHint(self, secret, guess):
        """
        :type secret: str
        :type guess: str
        :rtype: str
        """
        A = 0
        B = 0
        count = [0] * 10
        for i in range(len(secret)):
            if secret[i] == guess[i]:
                A += 1
            else:
                count[int(secret[i])] += 1
                if count[int(secret[i])] <= 0:
                    B += 1
                count[int(guess[i])] -= 1
                if count[int(guess[i])] >= 0:
                    B += 1
        return "%sA%sB" % (A, B)

File: BullsAndCows/BullsAndCows.cpp
#include<string>
#include<vector>
#include<algorithm>
using namespace std;

class Solution {
public:
	string getHint(string secret, string guess) {
		int aCnt = 0;
		int bCnt = 0;
		vector<int> sVec(10, 0);
		vector<int> gVec(10, 0);
		if (secret.size() != guess.size() || secret.empty())
			return "0A0B";
		for (int i = 0; i < secret.size(); i++) {
			char s = secret[i], g = guess[i];
			if (s == g) {
				aCnt++;
			}
			else {
				sVec[s - '0']++;
				gVec[g - '0']++;
			}
		}
		for (int i = 0; i < sVec.size(); i++) {
			bCnt += min(sVec[i], gVec[i]);
		}
		return to_string(aCnt) + 'A' + to_string(bCnt) + 'B';
	}
};